modulemd/modulemd-module-index.c

```diff
@@ -155,51 +155,10 @@ add_subdoc (ModulemdModuleIndex *self,
   switch (doctype)
     {
     case MODULEMD_YAML_DOC_PACKAGER:
-      if (modulemd_subdocument_info_get_mdversion (subdoc) <
-          MD_PACKAGER_VERSION_TWO)
-        {
-          g_set_error (error,
-                       MODULEMD_YAML_ERROR,
-                       MMD_YAML_ERROR_PARSE,
-                       "Invalid mdversion for a packager document");
-          return FALSE;
-        }
-
-      if (modulemd_subdocument_info_get_mdversion (subdoc) ==
-          MD_PACKAGER_VERSION_THREE)
-        {
-          packager = modulemd_packager_v3_parse_yaml (subdoc, error);
-
-          index =
-            modulemd_packager_v3_to_stream_v2_ext (packager, &nested_error);
-          if (!index)
-            {
-              g_propagate_error (error, g_steal_pointer (&nested_error));
-              return FALSE;
-            }
-
-          if (autogen_module_name)
-            {
-              /* nothing to do here since module/stream names were already
-               * auto-generated in order to add them to the index
-               */
-            }
-
-          /* merge index with override = FALSE and strict_default_streams = TRUE */
-          if (!modulemd_module_index_merge (
-                index, self, FALSE, TRUE, &nested_error))
-            {
-              g_propagate_error (error, g_steal_pointer (&nested_error));
-              return FALSE;
-            }
-
-          g_clear_object (&index);
-          break;
-        }
-
-      /* Fall through intentionally
-       * We will handle the v2 packager format below
-       */
+      g_warning (
+        "modulemd-packager document ignored while reading into a module "
+        "index");
```

Comment:
Don't issue a `g_warning()` here. This will call `assert()` if `G_DEBUG=fatal-warnings`.

Reply:
What we actually want to do here is treat this the same way we treat an unknown document type below: set the gerror and return `FALSE`.

Reply:
I implemented per our chat discussion: "If we see a PackagerV3 in the regular Index read, drop it on the floor (with a g_warning())". Apparently that's not what was intended.

I'll revise tomorrow.

Reply:
Sorry about that, I did indeed say "use `g_warning()`". I had forgotten that it `assert()`s and that what we really wanted to do was treat it the same as any other unknown subdocument.

Reply:
No worries. It should be better now.